main.go

```diff
@@ -277,6 +289,16 @@ func main() {
 
 	instanceName, _ := os.Hostname() // on an error, instanceName will be empty, which is ok
 
+	// A separate controller is needed because of hosted mode concerns.
+
+	nsSelReconciler := common.NamespaceSelectorReconciler{
+		Client: nsSelMgr.GetClient(),
```

Comment:
Could you please limit the fields in the namespaces that are cached by using the `Transform` function?